ui/src/app/core/photo-loader/photo-loader.service.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2020 the original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+import { map, switchMap } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+
+import { decode } from 'tiff';
+import TiffIfd from 'tiff/lib/tiffIfd';
+
+interface IImageData {

```

Comment:
please define the interface in its own file, where other interfaces are placed. Do not use 'I' prefix for interfaces, according to the angular style guide it is redundant